setup.py

```diff
@@ -99,41 +99,51 @@ def compile_native_invocation_handler(*possible_homes):
     # for android, we use SDL...
     LIBRARIES = ['sdl', 'log']
     LIBRARY_DIRS = ['libs/' + getenv('ARCH')]
+
 elif PLATFORM == 'darwin':
-    FRAMEWORK = subprocess.Popen(
+
+    JAVA_HOME = subprocess.Popen(
         '/usr/libexec/java_home',
         stdout=subprocess.PIPE, shell=True).communicate()[0]
+
     if PY3:
-        FRAMEWORK = FRAMEWORK.decode()
-    FRAMEWORK = FRAMEWORK.strip()
-    print('java_home: {0}\n'.format(FRAMEWORK))
-    if not FRAMEWORK:
+        JAVA_HOME = JAVA_HOME.decode()
```

Comment:
Decoding explicitly with `utf-8` seems as a better idea to me instead of relying on the defaults, even if this isn't section running on Windows.